### DISCUSSION TITLE
How to get initial form field values in handleSubmit?

### DISCUSSION THREAD

**Discussion by gajus** [react-hook-form/react-hook-form#6911] [Q&A]
I am adding analytics integration and I need to fire an event for each input which has a changed value upon submission, i.e. when form is submitted I need to identify which inputs have changed in value and what was the original / what is the current value.

The only option I see of how to do this is to use `watch` to monitor value of every input individually and then track a log changes using `useRef`. But perhaps there is a more straightforward path that does not require binding every input.

**Comment by lundn** [comment]
I guess you could leverage `dirtyFields` from formState to identify which fields were modified from their default values? I recon you have access to the default values somewhere in the form component?
